File: exam_portal_ui/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http: HttpClient) { }

  public getQuestionsofQuiz(qid: any) {
    return this._http.get(`${baseUrl}/question/quiz/all/${qid}`);
  }

  // get question for test
  public getQuestionOfQuizForTest(qid: any){
    return this._http.get(`${baseUrl}/question/quiz/${qid}`);
  }

  // add Question
  public addQuestion(question: any)
  {
    return this._http.post(`${baseUrl}/question/`, question);
  }

  // delete question
  public deleteQuestion(questionId: any){
    return this._http.delete(`${baseUrl}/question/${questionId}`)
  }

  // eval quiz
  public evalQuiz(questions: any){
    return this._http.post(`${baseUrl}/question/eval-quiz`, questions);
  }
}

File: examserver/src/main/java/com/exam/examserver/service/QuestionService.java
package com.exam.examserver.service;

import com.exam.examserver.entity.exam.Question;
import com.exam.examserver.entity.exam.Quiz;

import java.util.Set;

public interface QuestionService {
    public Question addQuestion(Question question);
    public Question updateQuestion(Question question);
    public Set<Question> getQuestions();
    public Question getQuestion(String questionId);
    public Set<Question> getQuestionsOfQuiz(Quiz quiz);

    public void deleteQuestion(String questionId);

    public Question getQuestionId(String questionId);
}
